src/components/blipTable/blipTable.component.ts

```diff
@@ -0,0 +1,133 @@
+import angular from 'core/angular';
+import { IRootElementService, IOnChangesObject, IScope } from 'angular';
+import * as uuid from 'uuid';
+import template from './BlipTableView.html';
+
+const BLIP_TABLE_PREFIX = 'blip-table-';
+
+/**
+ * Usage:
+ * <blip-table table-data="TableDataHere" selectable scrollable>
+ *
+      <column title="Column title here" width="Column width in percentage"
+        row-param="Row parameter on TableData here" sortable></column>
+
+      <column title="Column title here" column-class="Classes for header"
+        row-param="Row parameter on TableData here" row-class="Classes for row"
+        sort-by="Optional sorting parameter. If absent, row-param will be used instead" sortable></column>
+
+      <column title="Column title here"
+        row-param="Row parameter on TableData here" row-title="Optional row title on TableData here"></column>
+
+    </blip-table>
+ */
+class BlipTable {
+    public tableData: any[];
+    public columns: any[];
+    public elementId: string;
+    public scrollable: boolean;
+    public selectable: boolean;
+    public allChecked: boolean;
+    public selected: any[];
+
+    constructor(
+        private $element: IRootElementService,
+        private $scope: IScope,
+    ) {
+        this.elementId = `${BLIP_TABLE_PREFIX}${uuid.v4()}`;
+        this.columns = [];
+        this.scrollable = this.$element[0].hasAttribute('scrollable');
+        this.selectable = this.$element[0].hasAttribute('selectable');
+    }
+
+    $onInit() {
+        if (this.selectable) {
+            this.allChecked = false;
+            this.selected = [];
+
+            this.$scope.$watch('$ctrl.selected.length', (newVal: number) => {
+                if (newVal && newVal === this.tableData.length) {
+                    this.allChecked = true;
+                }
+            });
+        }
+    }
+
+    $onChanges(changesObj: IOnChangesObject) {
+        if (changesObj.tableData) {
+            this.tableData.forEach(el => {
+                if (el.checked === undefined) { el.checked = false; }
+            });
+            this.columns.forEach(c => c.resetSorting());
+
+            if (changesObj.tableData.previousValue.length === 0 && this.scrollable) {
+                this.setScrollHeight();
+            }
+        }
+    }
+
+    setScrollHeight() {
+        const scroller: HTMLDivElement = this.$element[0].querySelector('.bp-table-scroll-y-div');
+        scroller.style.maxHeight = `${scroller.offsetHeight}px`;
+    }
+
+    itemStateChange(state: boolean, $index: number) {
+        if (state === undefined) { return; }
+
+        const item = this.tableData[$index];
+        if (state) {
+            this.selected.push(item);
```

Comment:
Prefer immutable methods (`this.selected = this.selected.concat(item)`) to handle angular components properties. Mutable methods can lead to problems while handling binding mutations at `$onChanges` lifecycle method: https://docs.angularjs.org/guide/component